### PR TITLE
Revert "Revert "Brody/swap v2 e2e (#5915)" (#5987)"

### PR DESCRIPTION
**Pull request by BrodyHughes**
- The previous attempt to add this e2e messed up CI and caused it to loop forever bc of an error in the animation configs. I noticed this after it was merged unfortunately bc of weird timing with some other PRs it happened to pass. 
That issue is fixed here: https://github.com/rainbow-me/rainbow/pull/5988/commits/ef118ec8972ea38ddbbaf1433cb7eb7b415990e0

- The test was also failing bc of an issue with fetching hardhat balances and them not being set properly in userAssets. changes made in `userAssetsQueryFunction` now return hardhat assets correctly in the correct format and changes in `setUserAssets` make it more flexible in handling Maps and Arrays and checking both `native.balance.amount` and `balance.amount`. 
Those changes are here: https://github.com/rainbow-me/rainbow/pull/5988/commits/900979688589d94863e89665b373bd23880a424e

### CODE REVIEW

File: src/__swaps__/screens/Swap/hooks/useAssetsToSell.ts [hunk 1]
```diff
@@ -21,7 +22,11 @@ const sortBy = (by: UserAssetFilter) => {
 };
 
 export const useAssetsToSell = () => {
-  const { accountAddress: currentAddress, nativeCurrency: currentCurrency } = useAccountSettings();
+  const { accountAddress: currentAddress, nativeCurrency: currentCurrency, network: currentNetwork } = useAccountSettings();
+
+  const provider = getCachedProviderForNetwork(currentNetwork);
+  const providerUrl = provider?.connection?.url ?? '';
+  const connectedToHardhat = isHardHat(providerUrl);
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useAssetsToSell.ts:29]: I'd condense these to be the new method to do this in one line:

```
import { getIsHardhatConnected } from '@/handlers/web3';

const connectedToHardhat = getIsHardhatConnected();
```

Comment (BrodyHughes) [reply src/__swaps__/screens/Swap/hooks/useAssetsToSell.ts:29]: oops yeah

**Comment by brunobar79** [comment]
@BrodyHughes  conflicts 😭

File: src/state/assets/userAssets.ts [hunk 2]
```diff
@@ -270,9 +272,11 @@ export const userAssetsStore = createRainbowStore<UserAssetsState>(
       set(() => {
         const idsByChain = new Map<UserAssetFilter, UniqueId[]>();
         const unsortedChainBalances = new Map<ChainId, number>();
+        const isMap = userAssets instanceof Map;
+        const userAssetsArray = isMap ? Array.from(userAssets.values()) : userAssets;
 
-        userAssets.forEach(asset => {
-          const balance = Number(asset.native.balance.amount) ?? 0;
+        userAssetsArray.forEach(asset => {
+          const balance = Number(asset.native?.balance?.amount ?? asset.balance?.amount) ?? 0;
```
Comment (walmat) [inline src/state/assets/userAssets.ts:279]: I'm not sure if native balance will work just the same as `asset.balance`. This is something to look into.

**Review by walmat — APPROVED** [review]
Will need to expand the test suite quite a bit. But can be in a follow-up PR.

**Pull request merged**